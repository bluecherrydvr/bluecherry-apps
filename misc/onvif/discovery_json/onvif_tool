#!/usr/bin/env php
<?php

# test:
#$p = @popen("cat tests/onvif_tool.out", "r");
$p = @popen("/usr/lib/bluecherry/onvif_tool dummyaddr null null discover", "r");
if (!$p) {
	exit(1);
}
$wait_ip_line = true;
$cams_i = 0;
while ($str = fgets($p)) {

	//IP
	//scope strings
	//newline


	if ($str[0] == "\n" && !$wait_ip_line) {
		$wait_ip_line = true;
		$cams_i++;
		//$res[$cams_i] = $cam;
		print(json_encode($cam) . "\n");
	}

	if ($wait_ip_line){
		if (preg_match('/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/', $str, $match)) {
			$cam = Array();
			$cam['manufacturer'] = '';
			$cam['model_name'] = '';
			$cam['ipv4_port'] = '';
			//$cam['ipv6'] = '';
			//$cam['ipv6_path'] = '';
			//$cam['onvif'] = true;
			//$cam['exists'] = false;

			$cam['ipv4'] = $match[0];
			$cam['ipv4_path'] = 'http://'.$match[0].'/onvif/device_service';
			$cam['ipv4_port'] = $match[0].':80'; // FIXME might not be the case

			$wait_ip_line = false;
		}
	}else{
		//match "name/manufacturer"
		preg_match_all('#name/([\w/%\-]+)#', $str, $match);
		if (!empty($match[1]))
			$cam['manufacturer'] = urldecode($match[1][0]);
		//match hardware/modelname
		preg_match_all('#hardware/([\w/%\-]+)#', $str, $match);
		if (!empty($match[1]))
			$cam['model_name'] = $match[1][0];
	}
}
pclose($p);

//print(json_encode($res));

?>
