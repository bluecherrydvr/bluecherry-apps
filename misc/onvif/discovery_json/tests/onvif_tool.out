192.168.86.200
type/video_encoder
Profile/Streaming
Profile/G
type/audio_encoder
hardware/DS-2CD2742FWD-IZS
name/HIKVISION%20DS-2CD2742FWD-IZS
location/city/hangzhou

192.168.86.109
type/video_encoder
Profile/Streaming
Profile/G
type/ptz
hardware/DS-2CD4A35FWD-IZH
name/HIKVISION%20DS-2CD4A35FWD-IZH
location/city/hangzhou

192.168.86.91
location/country/America
name/LOREX
hardware/LNB8111B
Profile/Streaming
type/Network_Video_Transmitter
extension/unique_identifier

192.168.86.58
location/country/USA
name/Amcrest
hardware/IP8M-2496E
Profile/Streaming
type/Network_Video_Transmitter
extension/unique_identifier
Profile/T

192.168.86.72
location/country/china
name/Dahua
hardware/IPC-HDBW4431R-ZS
Profile/Streaming
type/Network_Video_Transmitter
extension/unique_identifier
Profile/Q/Operational

192.168.86.89
hardware/LNZ32P12
location/country/china
name/Lorex
type/video_encoder
type/ptz
type/audio_encoder
type/network_video_transmitter
Profile/Streaming

